src/tools.h

```diff
@@ -156,6 +156,10 @@ class CopyItem : public zim::writer::Item         //Article class that will be p
     {
        return std::unique_ptr<zim::writer::ContentProvider>(new ItemProvider(item));
     }
+
+    zim::writer::Hints getHints() const {
+      return {{ zim::writer::HintKeys::FRONT_ARTICLE, (item.getMimetype() == "text/html")}};
+    }
```

Comment:
@mgautierfr What about if the mimetype has additional attributes like `charset=utf8`? This won't be detected properly?

Reply:
I've missed that.
But in the same time, we probably don't want to put `text/html;raw=true` as front article.

Last push fix this.

---

On a more global way, as commented [here](https://github.com/kiwix/kiwix-lib/pull/403#issuecomment-671277426), we face the problem that we have to parse something we shouldn't. (either for `raw=true` than `text/html`)

Hopefully, the (next to implement) extra data feature will fix this.